apps/examples/mediaplayer/BufferInputDataSource.cpp

```diff
@@ -0,0 +1,76 @@
+/****************************************************************************
```

Comment:
Below is what we discussed, right?
```
Class app : InputdataSource
{
void main(){
    InputDataSource input = new InputDataSource();
    input.setSampleRate(16000);
   ...
    mediaPlayer.setDataSource(input)
}
/* Override
ssize_t read() 
{
   ADD CODE
}
```

Reply:
App overrides, `MediaPlayerObserverInterface` and `FocusChangerListener`
I think App inherit too many class